user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserQuotas.java

```diff
@@ -0,0 +1,201 @@
+     *
+     * @param username  Name of the user whose configuration changed
+     */
+    private void notifyChanges(String username) {
+        log.debug("Notifying changes for user {}", username);
+
+        ensurePath("/config/changes");
+
+        JsonObject json = new JsonObject().put("version", 2).put("entity_path", "users/" + username);
+        zkClient.createPersistentSequential("/config/changes/config_change_", json.encode().getBytes(Charset.defaultCharset()));
+    }
+
+    /**
+     * Ensures that the path in Zookeeper exists.
+     * It checks whether it already exists and in case it doesn't, it will create the path.
+     *
+     * @param path The Zookeeper path which should exist
+     */
+    private void ensurePath(String path)    {
+        if (!zkClient.exists(path))   {
+            zkClient.createPersistent(path, true);
+        }
+    }
+
+    /**
+     * Determine whether the given user has quotas.
+     *
+     * @param username Name of the user
+     *
+     * @return True if the user exists
+     */
+    public boolean exists(String username) {
+        byte[] data = zkClient.readData("/config/users/" + username, true);
+
+        if (data != null)   {
+            String jsonString = new String(data, Charset.defaultCharset());
+            JsonObject json = new JsonObject(jsonString);
+            validateJsonVersion(json);
+            JsonObject config = json.getJsonObject("config");
+
+            if (config != null) {
+                String prod = config.getString("producer_byte_rate");
+                String cons = config.getString("consumer_byte_rate");
+                String perc = config.getString("request_percentage");
+
+                if (prod != null || cons != null || perc != null) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Delete the quotas for the given user.
+     * It is not an error if the user doesn't exist, or doesn't currently have any quotas.
+     *
+     * @param username Name of the user
+     */
+    public void delete(String username) {
+        byte[] data = zkClient.readData("/config/users/" + username, true);
+
+        if (data != null)   {
+            log.debug("Deleting quotas for user {}", username);
+            zkClient.writeData("/config/users/" + username, deleteUserJson(data));
+            notifyChanges(username);
+        } else {
+            log.warn("Quotas for user {} already don't exist", username);
+        }
+    }
+
+    /**
+     * Deletes the quotas from existing JSON
+     *
+     * @param user JSON string with existing user configuration as byte[]
+     *
+     * @return  Returns the updated JSON without the quotas as byte array
+     */
+    protected byte[] deleteUserJson(byte[] user)   {
+        JsonObject json = new JsonObject(new String(user, Charset.defaultCharset()));
+
+        validateJsonVersion(json);
+
+        if (json.getJsonObject("config") == null)   {
```

Comment:
Do us all a favour and factor out a local var for `json.getJsonObject("config")`